Telink b91 Openthread Coprocessor support

yurvyn opened the pull request.

Commit:
dts: riscv: telink_b91: transform labels to uppercase

Transformed all dts labels to uppercase.

Signed-off-by: Yuriy Vynnychek <yura.vynnychek@telink-semi.com>

Commit:
samples: openthread: add tlsr9518adk board configuration

Added Telink B91 tlsr9518adk board support.

Signed-off-by: Yuriy Vynnychek <yura.vynnychek@telink-semi.com>

Commit:
drivers: serial: b91: full duplex mode support

Improved fifo_fill API: stop and return on RX.

Signed-off-by: Yuriy Vynnychek <yura.vynnychek@telink-semi.com>

Commit:
boards: riscv: tlsr9518adk80d: new "openthread" sample support

Added info about new "openthread" sample support.

Signed-off-by: Yuriy Vynnychek <yura.vynnychek@telink-semi.com>

rlubos commented on samples/net/openthread/coprocessor/overlay-tlsr9-br.conf:
```diff
@@ -0,0 +1,7 @@
+# Telink B91 TLSR9518ADK80D
```
I think it'd be better to add `boards/tlsr9518adk80d.conf` file to the sample - that way the board-specific configuration would be applied automatically, no need for an overlay.

yurvyn replied:
Done.  Created new `boards/tlsr9518adk80d.conf` file with the same content.

Please pay attention that this conf file is NOT loaded automatically.
With this sample I have to run cmake with `-DCONF_FILE="prj.conf overlay-rcp.conf"` and as I see the files from `boards` folder are not loaded automatically if `-DCONF_FILE` option is present.
Now I run with `-DCONF_FILE="prj.conf overlay-rcp.conf boards/tlsr9518adk80d.conf"`

rlubos replied:
If you specify a custom configuration explicitly with `-DCONF_FILE`, then indeed, that configuration takes precedence (by design). But if you operate on defaults (which is what we should aim for the samples), then the config file in boards directory should be merged with `prj.conf` automatically, when built for that board.

If you need to specify an overlay file to build with, please use `-DOVERLAY_CONFIG=overlay-rcp.conf` instead of overriding the entire configuration. In result, all 3 config files should be merged.

yurvyn replied:
Yes, it works. 
Thanks.

rlubos reviewed (commented):
Looks good, I'd drop the overlay file in favour of board-specific config.

Commit:
boards: riscv: tlsr9518adk80d: clean-up board documentation

- Fixe reversed UART TX/RX pins info
- Added info about CONFIG_FPU in case of Telink's toolchain

Signed-off-by: Yuriy Vynnychek <yura.vynnychek@telink-semi.com>

rlubos commented:
Shouldn't the configuration file rather be moved in commit https://github.com/zephyrproject-rtos/zephyr/pull/37578/commits/617e4a3a1d8cf484c80845bf1df4a5155e8b3a3f instead of https://github.com/zephyrproject-rtos/zephyr/pull/37578/commits/7109ae96ce44bff6df4d9b6ccd8d6050aeea24fb? It'd make more sense...

yurvyn commented:
> Shouldn't the configuration file rather be moved in commit [617e4a3](https://github.com/zephyrproject-rtos/zephyr/commit/617e4a3a1d8cf484c80845bf1df4a5155e8b3a3f) instead of [7109ae9](https://github.com/zephyrproject-rtos/zephyr/commit/7109ae96ce44bff6df4d9b6ccd8d6050aeea24fb)? It'd make more sense...

Done.

rlubos reviewed (approved):
Thanks

yurvyn commented:
One more approval is required.
Could anyone please approve the PR?

yurvyn commented:
@tbursztyka, 
Could you please approve the PR?
Thanks.

yurvyn commented:
@dcpleung , @katsuster, @mengxianglinx, @mgielda

One more approval is required.
Could you please approve the PR?
